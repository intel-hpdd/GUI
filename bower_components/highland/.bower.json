{
  "name": "highland",
  "homepage": "https://github.com/caolan/highland",
  "authors": [
    "Caolan McMahon"
  ],
  "description": "The high-level streams library.",
  "main": "dist/highland.js",
  "keywords": [
    "streams",
    "async",
    "events",
    "pipe"
  ],
  "license": "Apache 2.0",
  "ignore": [
    "**/.*",
    "Contributing.md",
    "Gruntfile.js",
    "bench",
    "docs",
    "node_modules",
    "bower_components",
    "tasks",
    "test.js",
    "test"
  ],
  "version": "2.7.1",
  "_release": "2.7.1",
  "_resolution": {
    "type": "version",
    "tag": "2.7.1",
    "commit": "197822ee483f1e38fd410863d0edb4ad07d06d4d"
  },
  "_source": "https://github.com/caolan/highland.git",
  "_target": "2.7.1",
  "_originalSource": "highland"
}