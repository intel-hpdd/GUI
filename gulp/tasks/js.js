//
// INTEL CONFIDENTIAL
//
// Copyright 2013-2016 Intel Corporation All Rights Reserved.
//
// The source code contained or described herein and all documents related
// to the source code ("Material") are owned by Intel Corporation or its
// suppliers or licensors. Title to the Material remains with Intel Corporation
// or its suppliers and licensors. The Material contains trade secrets and
// proprietary and confidential information of Intel or its suppliers and
// licensors. The Material is protected by worldwide copyright and trade secret
// laws and treaty provisions. No part of the Material may be used, copied,
// reproduced, modified, published, uploaded, posted, transmitted, distributed,
// or disclosed in any way without Intel's prior express written permission.
//
// No license under any patent, copyright, trade secret or other intellectual
// property right is granted to or conferred upon you by disclosure or delivery
// of the Materials, either expressly, by implication, inducement, estoppel or
// otherwise. Any license under such intellectual property rights must be
// express and approved by Intel in writing.

'use strict';

var paths = require('../paths.json');
var gulp = require('gulp');
var babel = require('gulp-babel');
var sourcemaps = require('gulp-sourcemaps');
var destDir = require('../dest-dir');
var relativeSourcemapsSource = require('gulp-relative-sourcemaps-source');

function jsDeps (fn) {
  return gulp.src(paths.js.deps, {
    since: gulp.lastRun(fn),
    base: '.'
  });
}

exports.jsDepsDev = function jsDepsDev () {
  return jsDeps(jsDepsDev)
  .pipe(gulp.dest('./dest'))
  .pipe(gulp.symlink('static/chroma_ui', { cwd: destDir }));
};

exports.jsDepsProd = function jsDepsProd () {
  return jsDeps(jsDepsProd)
  .pipe(gulp.dest('./dest'));
};

function socketWorker (fn) {
  return gulp.src(paths.js.socketWorker, {
    since: gulp.lastRun(fn)
  })
  .pipe(sourcemaps.init({ loadMaps: true }))
  .pipe(sourcemaps.write('.'));
}

exports.socketWorkerDev = function socketWorkerDev () {
  return socketWorker(socketWorkerDev)
  .pipe(gulp.dest('./dest'))
  .pipe(gulp.symlink('static/chroma_ui', { cwd: destDir }));
};

exports.socketWorkerProd = function socketWorkerProd () {
  return socketWorker(socketWorkerProd)
  .pipe(gulp.dest('./dest'))
  .pipe(gulp.dest('./dist'));
};

function systemConfig (fn) {
  return gulp.src(paths.js.config, {
    since: gulp.lastRun(fn),
    base: '.'
  });
}

exports.systemConfigDev = function systemConfigDev () {
  return systemConfig(systemConfigDev)
  .pipe(gulp.dest('./dest'))
  .pipe(gulp.symlink('static/chroma_ui', { cwd: destDir }));
};

var babelDev = babel.bind(null, {
  presets: [],
  babelrc: false,
  plugins: [
    'check-es2015-constants',
    'transform-flow-strip-types',
    'transform-class-properties',
    'transform-object-rest-spread',
    'transform-async-to-generator',
    'transform-es2015-classes',
    'transform-es2015-modules-systemjs'
  ]
});

var babelProd = babel.bind(null, {
  presets: ['es2015'],
  babelrc: false,
  plugins: [
    'transform-flow-strip-types',
    'transform-class-properties',
    'transform-object-rest-spread',
    [
      'angularjs-annotate',
      {
        explicitOnly: true
      }
    ],
    'transform-es2015-modules-systemjs'
  ]
});

function jsSource (fn) {
  return gulp.src(paths.js.source, {
    since: gulp.lastRun(fn),
    base: '.'
  })
  .pipe(sourcemaps.init());
}

exports.jsSourceDev = function jsSourceDev () {
  return jsSource(jsSourceDev)
  .pipe(babelDev())
  .pipe(sourcemaps.write({ sourceRoot: '/static/chroma_ui' }))
  .pipe(gulp.dest('./dest'))
  .pipe(gulp.symlink('static/chroma_ui', { cwd: destDir }));
};

exports.jsSourceProd = function jsSourceProd () {
  return jsSource(jsSourceProd)
  .pipe(babelProd())
  .pipe(relativeSourcemapsSource({dest: 'dest'}))
  .pipe(sourcemaps.write({ sourceRoot: '' }))
  .pipe(gulp.dest('./dest'));
};

exports.jsTest = function jsTest () {
  return gulp.src(paths.js.tests, {
    since: gulp.lastRun(jsTest),
    base: '.'
  })
  .pipe(sourcemaps.init())
  .pipe(babelDev())
  .pipe(sourcemaps.write({ sourceRoot: '' }))
  .pipe(gulp.dest('./dest'));
};

exports.jsTestFixtures = function jsTestFixtures () {
  return gulp.src(paths.js.testFixtures, {
    since: gulp.lastRun(jsTestFixtures),
    base: '.'
  })
  .pipe(gulp.dest('./dest'));
};

exports.jsTestDeps = function jsTestDeps () {
  return gulp.src(paths.js.testDeps, {
    since: gulp.lastRun(jsTestDeps),
    base: '.'
  })
  .pipe(gulp.dest('./dest'));
};
