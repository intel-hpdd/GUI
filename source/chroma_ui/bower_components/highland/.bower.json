{
  "name": "highland",
  "version": "2.4.0",
  "homepage": "https://github.com/caolan/highland",
  "authors": [
    "Caolan McMahon"
  ],
  "description": "The high-level streams library.",
  "main": "dist/highland.js",
  "keywords": [
    "streams",
    "async",
    "events",
    "pipe"
  ],
  "license": "Apache 2.0",
  "ignore": [
    "**/.*",
    "Contributing.md",
    "Gruntfile.js",
    "bench",
    "docs",
    "node_modules",
    "bower_components",
    "tasks",
    "test.js",
    "test"
  ],
  "_release": "2.4.0",
  "_resolution": {
    "type": "version",
    "tag": "2.4.0",
    "commit": "7ff0deb4abec773009324072d8855f394cc93467"
  },
  "_source": "https://github.com/caolan/highland.git",
  "_target": "2.4.0",
  "_originalSource": "highland"
}