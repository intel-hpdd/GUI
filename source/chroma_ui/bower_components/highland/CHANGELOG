Changelog
=========

This file does not aim to be comprehensive (you have git history for that),
rather it lists changes that might impact your own code as a consumer of
this library.

2.0.0
-----

* The `source.merge()` algorithm now evaluates the entire source stream before
  reading from all of the resulting streams in parallel (previously it would
  start reading as soon as the source emitted the next stream)
* The `merge()` function now attempts to balance inputs more fairly. For example,
  if stream A has 100 values buffered and stream B gets a new value after 100ms,
  if we read at 200ms we'll get a value from each stream. Previously it would
  exhaust the stream A buffer before reading from stream B.
