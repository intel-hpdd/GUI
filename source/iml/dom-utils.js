// @flow

type QuerySelector<T: HTMLElement> = (HTMLElement | Document, string) => T;

export const querySelector: QuerySelector<*> = (
  elm: HTMLElement | Document,
  sel: string
) => {
  const selected = elm.querySelector(sel);

  if (!selected)
    throw new Error(
      `Could not find selector ${sel} for element ${elm.nodeName}`
    );

  return selected;
};
