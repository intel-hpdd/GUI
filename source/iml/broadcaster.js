// @flow

//
// Copyright (c) 2017 Intel Corporation. All rights reserved.
// Use of this source code is governed by a MIT-style
// license that can be found in the LICENSE file.

import highland from 'highland';

import type {
  HighlandStreamT,
  errorWrapT
} from 'highland';


type streamFnT = () => HighlandStreamT<any>;

export default function broadcaster (source$:HighlandStreamT<any>):streamFnT {
  var latest:mixed;

  const viewers = [];

  source$
    .errors(error => {
      const err:errorWrapT = {
        __HighlandStreamError__: true,
        error
      };

      viewers.forEach(v => v.write(err));
    })
    .each((xs:mixed) => {
      latest = xs;

      viewers.forEach(v => v.write(xs));
    });

  const fn = () => {
    const viewer$:HighlandStreamT<any> = highland()
      .onDestroy(() => {
        const idx = viewers
          .indexOf(viewer$);

        if (idx !== -1)
          viewers.splice(idx, 1);
      });

    if (latest)
      viewer$.write(latest);

    viewers.push(viewer$);

    return viewer$;
  };

  fn.endBroadcast = () => source$.destroy();

  return fn;
}
