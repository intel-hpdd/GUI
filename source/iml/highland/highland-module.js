// @flow

//
// Copyright (c) 2018 DDN. All rights reserved.
// Use of this source code is governed by a MIT-style
// license that can be found in the LICENSE file.

import angular from 'angular';
import highland from 'highland';

export default angular
  .module('highland', [])
  .value('Î»', highland)
  .value('highland', highland).name;
