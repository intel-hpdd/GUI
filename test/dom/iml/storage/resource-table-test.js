// @flow

import Inferno from 'inferno';
import highland from 'highland';
import broadcaster from '../../../../source/iml/broadcaster.js';
import { renderToSnapshot } from '../../../test-utils.js';
import { querySelector } from '../../../../source/iml/dom-utils.js';

describe('ResourceTable DOM tests', () => {
  let ResourceTable, resourceClass, resources, alertIndicatorB, mockStore;

  beforeEach(() => {
    resourceClass = {
      class_name: 'HostNetworkInterface',
      columns: [
        { label: 'Host id', name: 'host_id' },
        { label: 'Name', name: 'name' },
        { label: 'Inet4 address', name: 'inet4_address' },
        { label: 'Inet4 prefix', name: 'inet4_prefix' },
        { label: 'Type', name: 'type' },
        { label: 'Up', name: 'up' }
      ],
      fields: [
        {
          class: 'Integer',
          label: 'Host id',
          name: 'host_id',
          optional: false,
          user_read_only: false
        },
        {
          class: 'String',
          label: 'Name',
          name: 'name',
          optional: false,
          user_read_only: false
        },
        {
          class: 'String',
          label: 'Inet4 address',
          name: 'inet4_address',
          optional: false,
          user_read_only: false
        },
        {
          class: 'Integer',
          label: 'Inet4 prefix',
          name: 'inet4_prefix',
          optional: false,
          user_read_only: false
        },
        {
          class: 'String',
          label: 'Type',
          name: 'type',
          optional: false,
          user_read_only: false
        },
        {
          class: 'Boolean',
          label: 'Up',
          name: 'up',
          optional: false,
          user_read_only: false
        }
      ],
      id: 14,
      label: 'linux_network-HostNetworkInterface',
      modified_at: '2017-07-19T13:58:03.413762',
      plugin_internal: true,
      plugin_name: 'linux_network',
      resource_uri: '/api/storage_resource_class/14/',
      user_creatable: false
    };

    resources = {
      meta: {
        limit: 10,
        next: '/api/storage_resource/?limit=10&offset=10',
        offset: 0,
        previous: null,
        total_count: 16
      },
      objects: [
        {
          alerts: [],
          alias: "HostNetworkInterface (10, u'enp0s10')",
          attributes: {
            host_id: {
              class: 'Integer',
              label: 'Host id',
              markup: '10',
              raw: 10
            },
            inet4_address: {
              class: 'ResourceReference',
              label: 'Inet4 address',
              markup: '<p>10.128.0.12</p>',
              raw: '/api/foo/1'
            },
            inet4_prefix: {
              class: 'Integer',
              label: 'Inet4 prefix',
              markup: '24',
              raw: 24
            },
            name: {
              class: 'String',
              label: 'Name',
              markup: 'enp0s10',
              raw: 'enp0s10'
            },
            type: { class: 'String', label: 'Type', markup: 'tcp', raw: 'tcp' },
            up: { class: 'Boolean', label: 'Up', markup: 'True', raw: true }
          },
          charts: [
            { series: ['tx_bytes'], title: 'tx_bytes' },
            { series: ['rx_bytes'], title: 'rx_bytes' }
          ],
          class_name: 'HostNetworkInterface',
          default_alias: "HostNetworkInterface (10, u'enp0s10')",
          deletable: false,
          id: 2,
          modified_at: '2017-07-19T14:23:30.678061',
          parent_classes: null,
          plugin_name: 'linux_network',
          propagated_alerts: [],
          resource_uri: '/api/storage_resource/2/',
          stats: {
            rx_bytes: {
              data: null,
              label: 'rx_bytes',
              name: 'rx_bytes',
              type: 'timeseries',
              unit_name: 'Bytes/s'
            },
            tx_bytes: {
              data: null,
              label: 'tx_bytes',
              name: 'tx_bytes',
              type: 'timeseries',
              unit_name: 'Bytes/s'
            }
          },
          storage_id_str: '[10, "enp0s10"]'
        },
        {
          alerts: [],
          alias: "HostNetworkInterface (10, u'enp0s3')",
          attributes: {
            host_id: {
              class: 'Integer',
              label: 'Host id',
              markup: '10',
              raw: 10
            },
            inet4_address: {
              class: 'String',
              label: 'Inet4 address',
              markup: '10.0.2.15',
              raw: '10.0.2.15'
            },
            inet4_prefix: {
              class: 'Integer',
              label: 'Inet4 prefix',
              markup: '24',
              raw: 24
            },
            name: {
              class: 'String',
              label: 'Name',
              markup: 'enp0s3',
              raw: 'enp0s3'
            },
            type: { class: 'String', label: 'Type', markup: 'tcp', raw: 'tcp' },
            up: { class: 'Boolean', label: 'Up', markup: 'True', raw: true }
          },
          charts: [
            { series: ['tx_bytes'], title: 'tx_bytes' },
            { series: ['rx_bytes'], title: 'rx_bytes' }
          ],
          class_name: 'HostNetworkInterface',
          default_alias: "HostNetworkInterface (10, u'enp0s3')",
          deletable: false,
          id: 3,
          modified_at: '2017-07-19T14:23:30.695521',
          parent_classes: null,
          plugin_name: 'linux_network',
          propagated_alerts: [],
          resource_uri: '/api/storage_resource/3/',
          stats: {
            rx_bytes: {
              data: null,
              label: 'rx_bytes',
              name: 'rx_bytes',
              type: 'timeseries',
              unit_name: 'Bytes/s'
            },
            tx_bytes: {
              data: null,
              label: 'tx_bytes',
              name: 'tx_bytes',
              type: 'timeseries',
              unit_name: 'Bytes/s'
            }
          },
          storage_id_str: '[10, "enp0s3"]'
        }
      ]
    };

    alertIndicatorB = broadcaster(highland([[]]));

    mockStore = jest.genMockFromModule(
      '../../../../source/iml/store/get-store.js'
    ).default;

    jest.mock('../../../../source/iml/store/get-store.js', () => mockStore);

    ({
      ResourceTable
    } = require('../../../../source/iml/storage/resource-table.js'));
  });

  it('should render without data', () => {
    resources.objects = [];

    expect(
      renderToSnapshot(
        <ResourceTable
          resourceClass={resourceClass}
          resources={resources}
          sortKey=""
          sortDesc={false}
          entries={10}
          alertIndicatorB={alertIndicatorB}
        />
      )
    ).toMatchSnapshot();
  });

  describe('with data', () => {
    let root;

    beforeEach(() => {
      root = document.createElement('div');
      querySelector(document, 'body').appendChild(root);

      Inferno.render(
        <ResourceTable
          resourceClass={resourceClass}
          resources={resources}
          sortKey=""
          sortDesc={false}
          entries={10}
          alertIndicatorB={alertIndicatorB}
        />,
        root
      );
    });

    afterEach(() => {
      querySelector(document, 'body').removeChild(root);
    });

    it('should render with data', () => {
      expect(root).toMatchSnapshot();
    });

    it('should dispatch a sort change', () => {
      querySelector(root, 'th a').click();

      expect(mockStore.dispatch).toHaveBeenCalledOnceWith({
        payload: { loading: true, sortDesc: true, sortKey: 'host_id' },
        type: 'SET_STORAGE_CONFIG'
      });
    });

    it('should dispatch an entries change', () => {
      querySelector(root, '.btn-info').click();
      root.querySelectorAll('li a')[1].click();

      expect(mockStore.dispatch).toHaveBeenCalledOnceWith({
        payload: { entries: 25, loading: true },
        type: 'SET_STORAGE_CONFIG'
      });
    });

    it('should dispatch an offset change', () => {
      querySelector(root, '.pagination-next a').click();

      expect(mockStore.dispatch).toHaveBeenCalledOnceWith({
        payload: { loading: true, offset: 10 },
        type: 'SET_STORAGE_CONFIG'
      });
    });
  });
});
