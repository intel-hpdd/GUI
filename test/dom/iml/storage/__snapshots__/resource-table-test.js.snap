// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResourceTable DOM tests should render without data 1`] = `
<div>
  <div
    class="well text-center resource-table-no-records"
  >
    <h1>
      No records found
    </h1>
  </div>
</div>
`;

exports[`ResourceTable DOM tests with data should render with data 1`] = `
<div>
  <div
    class="resource-table"
  >
    <div
      style="margin-bottom: 25px;"
    >
      <div
        class="btn-group dropdown "
      >
        <button
          class="btn btn-info btn-sm dropdown-toggle"
          type="button"
          aria-haspopup="true"
          aria-expanded="false"
        >
          Entries: 
          10
          <span
            class="caret"
          />
        </button>
        <ul
          class="dropdown-menu"
          role="menu"
        >
          <li>
            <a>
              10
            </a>
          </li>
          <li>
            <a>
              25
            </a>
          </li>
          <li>
            <a>
              50
            </a>
          </li>
          <li>
            <a>
              100
            </a>
          </li>
        </ul>
      </div>
    </div>
    <table
      class="table"
    >
      <tr>
        <th>
          Name
        </th>
        <th>
          Alerts
        </th>
        <th>
          <a>
            Host id
            <i
              class="fa "
            />
          </a>
        </th>
        <th>
          <a>
            Name
            <i
              class="fa "
            />
          </a>
        </th>
        <th>
          <a>
            Inet4 address
            <i
              class="fa "
            />
          </a>
        </th>
        <th>
          <a>
            Inet4 prefix
            <i
              class="fa "
            />
          </a>
        </th>
        <th>
          <a>
            Type
            <i
              class="fa "
            />
          </a>
        </th>
        <th>
          <a>
            Up
            <i
              class="fa "
            />
          </a>
        </th>
      </tr>
      <tbody>
        <tr>
          <td>
            <a
              href="about:blankconfigure/storage/2"
            >
              HostNetworkInterface (10, u'enp0s10')
            </a>
          </td>
          <td>
            <span
              class="record-state"
            >
              <span
                style="position: relative;"
              >
                <span
                  class="icon-wrap tooltip-container tooltip-hover"
                  popoverbutton="true"
                >
                  <i
                    class="fa activate-popover fa-check-circle"
                  />
                  <div
                    class="tooltip inferno-tt top small "
                  >
                    <div
                      class="tooltip-arrow"
                    />
                    <div
                      class="tooltip-inner"
                    >
                      <span>
                        No Issues
                      </span>
                    </div>
                  </div>
                </span>
              </span>
              <span
                class="state-label"
                style="padding-left: 10px;"
              >
                No Issues
              </span>
            </span>
          </td>
          <td>
            <span>
              10
            </span>
          </td>
          <td>
            <span>
              enp0s10
            </span>
          </td>
          <td>
            <a
              href="about:blankconfigure/storage/1"
            >
              <p>
                10.128.0.12
              </p>
            </a>
          </td>
          <td>
            <span>
              24
            </span>
          </td>
          <td>
            <span>
              tcp
            </span>
          </td>
          <td>
            <span>
              True
            </span>
          </td>
        </tr>
        <tr>
          <td>
            <a
              href="about:blankconfigure/storage/3"
            >
              HostNetworkInterface (10, u'enp0s3')
            </a>
          </td>
          <td>
            <span
              class="record-state"
            >
              <span
                style="position: relative;"
              >
                <span
                  class="icon-wrap tooltip-container tooltip-hover"
                  popoverbutton="true"
                >
                  <i
                    class="fa activate-popover fa-check-circle"
                  />
                  <div
                    class="tooltip inferno-tt top small "
                  >
                    <div
                      class="tooltip-arrow"
                    />
                    <div
                      class="tooltip-inner"
                    >
                      <span>
                        No Issues
                      </span>
                    </div>
                  </div>
                </span>
              </span>
              <span
                class="state-label"
                style="padding-left: 10px;"
              >
                No Issues
              </span>
            </span>
          </td>
          <td>
            <span>
              10
            </span>
          </td>
          <td>
            <span>
              enp0s3
            </span>
          </td>
          <td>
            <span>
              10.0.2.15
            </span>
          </td>
          <td>
            <span>
              24
            </span>
          </td>
          <td>
            <span>
              tcp
            </span>
          </td>
          <td>
            <span>
              True
            </span>
          </td>
        </tr>
      </tbody>
    </table>
    <div
      class="text-center"
    >
      <ul
        class="pagination"
      >
        <li
          class="pagination-prev disabled"
        >
          <a>
            ‹
          </a>
        </li>
        <li
          class="pagination-page active"
        >
          <a>
            1
          </a>
        </li>
        <li
          class="pagination-page "
        >
          <a>
            2
          </a>
        </li>
        <li
          class="pagination-next "
        >
          <a>
            ›
          </a>
        </li>
      </ul>
    </div>
    <div
      class="text-center"
      style="color: rgb(153, 153, 153);"
    >
      <span>
        showing 
        1
        -
        10
        of 
        16
      </span>
    </div>
  </div>
</div>
`;
